### DISCUSSION TITLE
MCP4725 code won't compile

### DISCUSSION THREAD

**Discussion by jamesluyten** [community/community#65782] [Programming Help]
Hi - trying to use MCP4725 code - triangle from Library - won't compile on Arduino MKR1010 gives following errors: Please help!

/Users/jamesluyten/Documents/Arduino/libraries/Adafruit_MCP4725/Adafruit_MCP4725 2.cpp:44:6: error: prototype for 'void Adafruit_MCP4725::begin(uint8_t)' does not match any in class 'Adafruit_MCP4725'
 void Adafruit_MCP4725::begin(uint8_t addr) {
      ^~~~~~~~~~~~~~~~
In file included from /Users/jamesluyten/Documents/Arduino/libraries/Adafruit_MCP4725/Adafruit_MCP4725 2.cpp:29:0:
/Users/jamesluyten/Documents/Arduino/libraries/Adafruit_MCP4725/Adafruit_MCP4725.h:28:8: error: candidate is: bool Adafruit_MCP4725::begin(uint8_t, arduino::TwoWire*)
   bool begin(uint8_t i2c_address = MCP4725_I2CADDR_DEFAULT,
        ^~~~~
/Users/jamesluyten/Documents/Arduino/libraries/Adafruit_MCP4725/Adafruit_MCP4725 2.cpp:65:6: error: prototype for 'void Adafruit_MCP4725::setVoltage(uint16_t, bool)' does not match any in class 'Adafruit_MCP4725'
 void Adafruit_MCP4725::setVoltage( uint16_t output, bool writeEEPROM )
      ^~~~~~~~~~~~~~~~
In file included from /Users/jamesluyten/Documents/Arduino/libraries/Adafruit_MCP4725/Adafruit_MCP4725 2.cpp:29:0:
/Users/jamesluyten/Documents/Arduino/libraries/Adafruit_MCP4725/Adafruit_MCP4725.h:30:8: error: candidate is: bool Adafruit_MCP4725::setVoltage(uint16_t, bool, uint32_t)
   bool setVoltage(uint16_t output, bool writeEEPROM,
        ^~~~~~~~~~
exit status 1   Compilation error: exit status 1
code:/*! 
    @file     trianglewave.pde
    @author   Adafruit Industries
    @license  BSD (see license.txt)

    This example will generate a triangle wave with the MCP4725 DAC.   

    This is an example sketch for the Adafruit MCP4725 breakout board
    ----> http://www.adafruit.com/products/935
 
    Adafruit invests time and resources providing this open source code, 
    please support Adafruit and open-source hardware by purchasing 
    products from Adafruit!
*/
/**************************************************************************/
#include <Wire.h>
#include <Adafruit_MCP4725.h>

Adafruit_MCP4725 dac;

void setup(void) {
  Serial.begin(9600);
  Serial.println("Hello!");

  // For Adafruit MCP4725A1 the address is 0x62 (default) or 0x63 (ADDR pin tied to VCC)
  // For MCP4725A0 the address is 0x60 or 0x61
  // For MCP4725A2 the address is 0x64 or 0x65
  dac.begin(0x62);
    
  Serial.println("Generating a triangle wave");
}

void loop(void) {
    uint32_t counter;
    // Run through the full 12-bit scale for a triangle wave
    for (counter = 0; counter < 4095; counter++)
    {
      dac.setVoltage(counter, false);
    }
    for (counter = 4095; counter > 0; counter--)
    {
      dac.setVoltage(counter, false);
    }
}

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @jamesluyten ! 

We’ve moved your post to our new Programming Help 🧑‍💻 category, which is more appropriate for this type of discussion.

Please review our [guidelines](https://github.com/orgs/community/discussions/51384) about the Programming Help category for more information.

**Comment by jamesluyten** [comment]
Thanks!
Apologies for the speeling Jim Luyten +1.508.566.1198
